File: Assets/Bundles/Server/Resources/TS/Services/Global/Chat/Commands/Match/MatchStartCommand.lua
-- Compiled with unity-ts v2.1.0-75
local Flamework = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Flamework
local Game = require("Shared/TS/Game").Game
local ColorUtil = require("Shared/TS/Util/ColorUtil").ColorUtil
local Task = require("Shared/TS/Util/Task").Task
local Theme = require("Shared/TS/Util/Theme").Theme
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local StartMatchCommand
do
	local super = ChatCommand
	StartMatchCommand = setmetatable({}, {
		__tostring = function()
			return "StartMatchCommand"
		end,
		__index = super,
	})
	StartMatchCommand.__index = StartMatchCommand
	function StartMatchCommand.new(...)
		local self = setmetatable({}, StartMatchCommand)
		return self:constructor(...) or self
	end
	function StartMatchCommand:constructor()
		super.constructor(self, "start")
	end
	function StartMatchCommand:Execute(player, args)
		-- Start match when match is ready.
		Task:Spawn(function()
			(Flamework.resolveDependency("Bundles/Server/Services/Match/MatchService@MatchService")):WaitForMatchReady();
			(Flamework.resolveDependency("Bundles/Server/Services/Match/MatchService@MatchService")):StartMatch()
			Game:BroadcastMessage(ColorUtil:ColoredText(player.username, Theme.Aqua) .. " started the match!")
		end)
	end
end
return {
	StartMatchCommand = StartMatchCommand,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Server/Services/Global/Chat/Commands/Match/MatchStartCommand.ts
import { Dependency } from "@easy-games/flamework-core";
import { ChatCommand } from "@Easy/Core/Shared/Commands/ChatCommand";
import { Game } from "@Easy/Core/Shared/Game";
import { Player } from "@Easy/Core/Shared/Player/Player";
import { ColorUtil } from "@Easy/Core/Shared/Util/ColorUtil";
import { Task } from "@Easy/Core/Shared/Util/Task";
import { Theme } from "@Easy/Core/Shared/Util/Theme";
import { MatchService } from "Server/Services/Match/MatchService";

export class MatchStartCommand extends ChatCommand {
	constructor() {
		super("start");
	}

	public Execute(player: Player, args: string[]): void {
		/* Start match when match is ready. */
		Task.Spawn(() => {
			Dependency<MatchService>().WaitForMatchReady();
			Dependency<MatchService>().StartMatch();
			Game.BroadcastMessage(`${ColorUtil.ColoredText(Theme.aqua, player.username)} started the match!`);
		});
	}
}
